account_financial_risk/models/res_partner.py

```diff
@@ -0,0 +1,554 @@
+
+    @api.onchange("credit_currency")
+    def _onchange_credit_currency(self):
+        for partner in self:
+            if partner.credit_currency == "manual":
+                partner.manual_credit_currency_id = partner.risk_currency_id
+            else:
+                partner.manual_credit_currency_id = False
+
+    @api.onchange("risk_currency_id")
+    def _onchange_risk_currency_id(self):
+        self._compute_risk_account_amount()
+
+    @api.constrains("credit_currency", "manual_credit_currency_id")
+    def _check_credit_currency(self):
+        for partner in self:
+            if (
+                partner.credit_currency == "manual"
+                and not partner.manual_credit_currency_id
+            ):
+                raise ValidationError(_("Choose Manual Credit Currency."))
+
+    def _compute_risk_allow_edit(self):
+        self.update(
+            {
+                "risk_allow_edit": self.env.user.has_group(
+                    "account_financial_risk.group_account_financial_risk_manager"
+                )
+            }
+        )
+
+    @api.model
+    def _get_risk_company_domain(self):
+        return [("company_id", "in", self.env.companies.ids)]
+
+    def _get_field_risk_model_domain(self, field_name):
+        """Returns a tuple with model name and domain"""
+        risk_account_groups = self._risk_account_groups()
+        if field_name == "risk_invoice_draft":
+            domain = risk_account_groups["draft"]["domain"]
+        elif field_name.endswith("_unpaid"):
+            domain = risk_account_groups["unpaid"]["domain"]
+        else:
+            domain = risk_account_groups["open"]["domain"]
+        # Usually this method is called in form view (one record in self)
+        account_receivable_id = self[:1].property_account_receivable_id.id
+        # Partner receivable account determines if amount is in invoice field
+        if field_name != "risk_invoice_draft":
+            if field_name.startswith("risk_invoice_"):
+                domain.append(("account_id", "=", account_receivable_id))
+            else:
+                domain.append(("account_id", "!=", account_receivable_id))
+        domain.append(("partner_id", "in", self.ids))
+        return "account.move.line", domain
+
+    @api.model
+    def _risk_account_groups(self):
+        max_date = self._max_risk_date_due()
+        company_domain = self._get_risk_company_domain()
+        fields = [
+            "amount_residual:sum",
+            "amount_residual_currency:sum",
+        ]
+        groupby = ["partner_id", "account_id", "currency_id"]
+        return {
+            "draft": {
+                "domain": company_domain
+                + [
+                    ("move_id.move_type", "in", ["out_invoice", "out_refund"]),
+                    ("account_type", "=", "asset_receivable"),
+                    ("parent_state", "in", ["draft", "proforma", "proforma2"]),
```

Comment:
I think these states are obsolete and must be updated.

```suggestion
                    ("parent_state", "=", "draft"),
```

Reply:
done